common/source/fs/minixfs/MinixFSSuperblock.cpp

```diff
@@ -300,7 +298,8 @@ void MinixFSSuperblock::readBlocks(uint16 block, uint32 num_blocks, char* buffer
   assert(fread(buffer, 1, BLOCK_SIZE * num_blocks, (FILE*)s_dev_) == BLOCK_SIZE * num_blocks);
 #else
   BDVirtualDevice* bdvd = BDManager::getInstance()->getDeviceByNumber(s_dev_);
-  bdvd->readData(block * bdvd->getBlockSize(), num_blocks * bdvd->getBlockSize(), buffer);
+  int32 read_retval = bdvd->readData(block * bdvd->getBlockSize(), num_blocks * bdvd->getBlockSize(), buffer);
+  assert(read_retval != -1 && "Disk read failed in MinixFSSuperblock::readBlocks()");
```

Comment:
Check that retval == read/write size instead
https://github.com/Woazboat/sweb/commit/597f6a04b5ff39551850f4cff0750c7be312a940